<launch>

  <node pkg="dronedoc" type="odom_publisher" name="odom_publisher" respawn="false" output="screen" />

  <node pkg="dronedoc" type="px4_vel_controller" name="px4_vel_controller">
    <param name="origin_frame" value="map"/>
    <param name="vehicle_frame" value="base_link"/>
    <param name="move_base_topic" value="/cmd_vel"/>
    <param name="mavros_topic" value="/mavros/setpoint_velocity/cmd_vel"/>
  </node>

  <node pkg="move_base" type="move_base" name="move_base" respawn="false" output="screen">

    <!-- update frequency of global plan -->
    <param name="planner_frequency" value="2.0"/>

    <!-- Common params for costmap -->
    <rosparam command="load" ns="global_costmap" file="$(find dronedoc)/config/costmap_common_params.yaml"/>
    <rosparam command="load" ns="local_costmap" file="$(find dronedoc)/config/costmap_common_params.yaml"/>

    <!-- Params for global costmap -->
    <rosparam command="load" file="$(find dronedoc)/config/global_costmap_params.yaml"/>

    <!-- Params for local costmap -->
    <rosparam command="load" file="$(find dronedoc)/config/local_costmap_params.yaml"/>

    <!-- Params for local planner -->
    <rosparam command="load" file="$(find dronedoc)/config/base_local_planner_params.yaml"/>
  </node>

</launch>